src/libcollections/binary_heap.rs

```diff
@@ -967,6 +974,13 @@ pub struct Iter<'a, T: 'a> {
     iter: slice::Iter<'a, T>,
 }
 
+#[stable(feature = "collection_debug", since = "1.15.0")]
+impl<'a, T: 'a> fmt::Debug for Iter<'a, T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad("BinaryHeap::Iter { .. }")
```

Comment:
I kept the implementation as easy as possible. Also, why would it need an implementation for `T: fmt::Debug`? Except if you want to see the internals, I'm not sure this is very useful.